internal/bminventory/inventory.go

```diff
@@ -555,6 +555,10 @@ func (b *bareMetalInventory) RegisterCluster(ctx context.Context, params install
 		cluster.SSHPublicKey = sshPublicKey
 	}
 
+	if _, err = b.authHandler.Client.AccountsMgmt.CreateSubscription(ctx, id); err != nil {
```

Comment:
try adding the ocmClient to NewBareMetalInventory, i.e. no need to change auth_handler.
see: https://github.com/openshift/assisted-service/blob/92e3dada0586054039cb53f468635c1b3501f01a/cmd/main.go#L290

Reply:
done.